### PR TITLE
Handle SharpRaven deprecation

### PR DESCRIPTION
**Pull request by huoyaoyuan**
SharpRaven is [deprecated](https://www.nuget.org/packages/SharpRaven/2.4.0) by its owner. Switching to supported new SDK.

Note: the new SDK is using much more static members. The instance members actually used are internal. Since we are subscribing to static event, it feels OK to me.

### CODE REVIEW

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public class OsuGame : OsuGameBase, IKeyBindingHandler<GlobalAction>
         [Cached]
         private readonly ScreenshotManager screenshotManager = new ScreenshotManager();
 
-        protected RavenLogger RavenLogger;
+        protected SentryLogger RavenLogger;
```
Comment (bdach) [inline osu.Game/OsuGame.cs]: Member can be renamed too.

File: osu.Game/Utils/SentryLogger.cs [hunk 2]
```diff
@@ -81,19 +81,9 @@ private bool shouldSubmitException(Exception exception)
...
-            {
-                lock (tasks)
-                    tasks.Remove(task);
-            });
-        }
-
         #region Disposal
 
-        ~RavenLogger()
+        ~SentryLogger()
```
Comment (bdach) [inline osu.Game/Utils/SentryLogger.cs]: Does this finalizer even need to exist?

File: osu.Game/Utils/SentryLogger.cs [hunk 3]
```diff
@@ -112,7 +102,8 @@ protected virtual void Dispose(bool isDisposing)
                 return;
 
             isDisposed = true;
-            lock (tasks) Task.WaitAll(tasks.ToArray(), 5000);
+            sentry.Dispose();
```
Comment (bdach) [inline osu.Game/Utils/SentryLogger.cs]: I'd probably guard this for null despite `isDisposed` anyways, if this is going to be called in the finalizer - no guarantees of anything there. Better safe than sorry with these things.

File: osu.Game/Utils/SentryLogger.cs [hunk 4]
```diff
@@ -7,26 +7,26 @@
...
 namespace osu.Game.Utils
 {
     /// <summary>
     /// Report errors to sentry.
     /// </summary>
-    public class RavenLogger : IDisposable
+    public class SentryLogger : IDisposable
     {
-        private readonly RavenClient raven = new RavenClient("https://5e342cd55f294edebdc9ad604d28bbd3@sentry.io/1255255");
+        private IDisposable sentry;
```
Comment (bdach) [inline osu.Game/Utils/SentryLogger.cs]: Weird API choice *to have `IDisposable`* as the returned type, but alas...

Comment (huoyaoyuan) [reply osu.Game/Utils/SentryLogger.cs]: There is a good sense `SentryClient`, but missing `Breadcrumb` support.
An alternative is use a permanent scope.

**Review by peppy — APPROVED** [review]
Looks sane

**Pull request merged**